src/components/blocks.ts

```diff
@@ -143,6 +145,32 @@ export default class Blocks {
     this.blocks[first] = secondBlock;
   }
 
+  /**
+   * Move a block from one to another index
+   * @param {Number} toIndex - new index of the block
+   * @param {Number} fromIndex - block to move
+   */
+  public move(toIndex: number, fromIndex: number): void {
+    const block = this.blocks.splice(fromIndex, 1)[0];
+
+    // manipulate DOM
+    const prevIndex = toIndex - 1;
+    const previousBlockIndex = prevIndex < 0 ? 0 : prevIndex;
+    const previousBlock = this.blocks[previousBlockIndex];
+    if (toIndex > 0) {
+      this.insertToDOM(block, 'afterend', previousBlock);
+    } else {
+      this.insertToDOM(block, 'beforebegin', previousBlock);
+    }
+
+    // move in array
+    this.blocks.splice(toIndex, 0, block);
+
+    // invoke hook
+    const event: MoveEvent = { fromIndex, toIndex };
+    block.call(BlockToolAPI.MOVED, event);
```

Comment:
This is ok as well:
```
block.call(BlockAPI.MOVED, { fromIndex, toIndex });
```

Reply:
That's right, 👍 I just wanted to have the type safety here.